src/fides/api/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -69,26 +72,46 @@ def create_consent_request(
             "Application redis cache required, but it is currently disabled! Please update your application configuration to enable integration with a redis cache."
         )
 
-    if not data.email:
-        raise HTTPException(HTTP_400_BAD_REQUEST, detail="An email address is required")
-
-    identity = ProvidedIdentity.filter(
-        db=db,
-        conditions=(
-            (ProvidedIdentity.field_name == ProvidedIdentityType.email)
-            & (ProvidedIdentity.hashed_value == ProvidedIdentity.hash_value(data.email))
-            & (ProvidedIdentity.privacy_request_id.is_(None))
-        ),
-    ).first()
+    if not data.email and not data.phone_number:
+        raise HTTPException(
+            HTTP_400_BAD_REQUEST,
+            detail="An email address or phone number identity is required",
+        )
 
-    if not identity:
-        provided_identity_data = {
-            "privacy_request_id": None,
-            "field_name": "email",
-            "hashed_value": ProvidedIdentity.hash_value(data.email),
-            "encrypted_value": {"value": data.email},
-        }
-        identity = ProvidedIdentity.create(db, data=provided_identity_data)
+    # consent requests, unlike privacy requests, only accept 1 identity type- email or phone
+    # if both identity types are provided, use identity type if defined in CONFIG.notifications.notification_service_type, else default to email
+    if data.email and data.phone_number:
+        messaging_method = get_messaging_method(
+            CONFIG.notifications.notification_service_type
+        )
+        if messaging_method == MessagingMethod.EMAIL:
+            identity = _get_or_create_provided_identity(
+                db=db,
+                identity_data=data,
+                identity_type=ProvidedIdentityType.email.value,
+            )
+        elif messaging_method == MessagingMethod.SMS:
+            identity = _get_or_create_provided_identity(
+                db=db,
+                identity_data=data,
+                identity_type=ProvidedIdentityType.phone_number.value,
+            )
+        else:
+            identity = _get_or_create_provided_identity(
+                db=db,
+                identity_data=data,
+                identity_type=ProvidedIdentityType.email.value,
+            )
+    elif data.email:
```

Comment:
this may be a bit deeper than we want to go in this PR, but i think there's an edge case here that i'm not sure we're handling very cleanly. specifically, if the client (whether it's the privacy center UI or theoretically some other API client) here submits _only_ an `email` or `phone_number` while their app is configured for an incompatible notification service  -- e.g. if their `notification_service_type` property is set to `twilio_text` but the request here contains only an `email`.

i think/assume we'll probably end up erroring out much further downstream in the messaging dispatcher - but i think we could probably be a bit more defensive here (and any other endpoints that go through similar dispatching logic), and just cut things off before we get that far to give what's likely a clearer error message to the user (probably a `422` here?)

but maybe i'm missing something. either way, i don't think it's a must-have at the moment, but curious to get your thoughts.

Reply:
@adamsachs you're correct that in this use case:

1. A customer wants to enable notifications
2. Their `notification_service_type` set to `twilio_text`
3. Their privacy-center `config.json` allows for only email to be sent via the API
4. A privacy request OR consent request is submitted with only an email identity
5. The app will raise a `500` with message ` Error sending the verification code message: No phone identity supplied.`. 

I'd prefer to keep this logic within the messaging service, since it's a central place to handle all things messaging and erroring. Otherwise, we'll have to implement the checks on each caller, which will be more of a hassle to manage (e.g. if we implement a new caller, we have to remember to implement this check).

For now, I'll open a new follow-up ticket to do the following:

1. move this check earlier on in the messaging service, and
2. add new Exception type to the dispatcher to raise `422` code instead, using this new type to handle this err case, and 
3. catch this new exception in every place where we call our message dispatcher, and
2. update the message to be clearer: ` Error sending the verification code message: Message service expecting sms identity, but no sms identity was supplied.

Reply:
ticket here- https://github.com/ethyca/fides/issues/2106

Reply:
nice, your point about keeping the logic within the messaging service does make sense. and so moving the check earlier in the messaging service makes a lot of sense to me.

it's a detail, but i'm not sure if i like raising the `422` from within the dispatch service - but i do get that you want to distinguish it as a specific type of exception and not just a generic `MessageDispatchException`, so you can then handle it appropriately in the places that use the dispatcher (which i also think is a great approach). so maybe just another subtype of `MessageDispatchException` that clearly indicates what's going wrong here, so it can be properly handled by the endpoint code and reported as a `422`? i guess i'm just a bit worried that if we introduce `HTTPException`s into the `message_dispatch_service`, it breaks the separation of concerns. feels a bit cleaner to have the underlying service just describe what went wrong, independent of the HTTP endpoint and how it's called; and then the specific pieces of endpoint code figure out which `HTTPException` to throw back to the user based on what's thrown up by the service underneath.

anyway, i'm splitting hairs a bit here, so feel free to ignore :) in general i think where you landed makes a lot of sense!

Reply:
ah I didn't mean literally raise the 422 as an `HTTPException`, but instead use a different error class, maybe `MessageDispatchValueException` so that the endpoint can handle throwing 422. 

But I like the idea of using a subtype of `MessageDispatchException` instead of a new Exception class altogether.

Reply:
But yes, you're totally right! We shouldn't be involving HTTP concepts in the messaging service

Reply:
updated requirements in ticket `add new Exception type (perhaps a subtype of `MessageDispatchException`) to the dispatcher such that the endpoint can catch and raise 422 code`

Reply:
ah ok great, we're totally on the same page then  👍